### ISSUE TITLE
Tracking issue: Improved ergonomics for defining constraints in AirScript

### ISSUE THREAD

**Issue by grjte** [0xPolygonMiden/air-script#77]
### Goal(s)
- better ergonomics when defining constraints in AirScript

### Details

To improve the ergonomics of the AirScript language, we should add support for:

- defining intermediate variables with the `let` keyword with these 3 types: scalars, vectors, matrices
- declaring named constants with these 3 types: scalars, vectors, matrices
- grouping columns (e.g.: a[4]) for
    - declarations in the `trace_columns` section
    - codegen referencing groups of columns (e.g. for generating one of the [structs Polygon Zero uses](https://github.com/mir-protocol/plonky2/blob/main/evm/src/cpu/columns/mod.rs) when defining constraints)

```[tasklist]

### Must have

- [x] #61 
- [x] #60 
- [x] #83
- [x] #85
```

```[tasklist]

### Nice to have

- [ ] #87
- [ ] #86
- [x] #89
- [x] #26
```

## Working group:
@Al-Kindi-0, @grjte, @Overcastan, @tohrnii 

<details>
<summary>Workflow</summary>
    
- Discussion should happen here or in the related sub-issues.
- PRs should only be merged by the coordinator, to ensure everyone is able to review.
- Aim to complete reviews within 24 hours.
- When a related sub-issue is opened:
    - add it to the list of sub-issues in this tracking issue
- When opening a related PR:
    - request review from everyone in this working group
- When a sub-issue is completed:
    - close the related issue with a comment that links to the PR where the work was completed
    
</details>

### Coordinator: @tohrnii
<details>
    
The working group coordinator ensures scope & progress tracking are transparent and accurate. They will:

- Merge approved PRs after all working group members have completed their reviews.
    - add the PR # to the relevant section of the current tracking PR.
    - close any completed sub-issue(s) with a comment that links to the PR where the work was completed
- Monitor workflow items and complete anything that slips through the cracks.
- Monitor scope to see if anything is untracked or unclear. Create missing sub-issues or initiate discussion as required.
- Monitor progress to see if there's anything which isn't moving forward. Initiate discussion as required.
- Identify PRs with especially significant changes and add @grjte and @bobbinth for review.
</details>

**Comment by grjte** [comment]
Do we want to use column-major or row-major form for matrices? (https://github.com/0xPolygonMiden/air-script/pull/63#issuecomment-1327417491) This affects many of the sub-issues here.

@bobbinth what do you think?

**Comment by bobbinth** [comment]
Ultimately, I don't think it matters since we can always transform one form to the other outside of the file. However, since we have to pick something, I think we should go with w/e is convenient for our current use cases.

Currently, the only use case for matrix constants is MDS matrix multiplication in Rescue constraints. There, I believe we need to iterate over rows to perform matrix-vector multiplication. So, I think this means that we should go with row-major form.

**Comment by Dominik1999** [comment]
Any progress here? Do you need anything to be unblocked?

**Comment by tohrnii** [comment]
@Dominik1999 Issues 60 and 61 are currently being worked on and will most likely be completed this week and 83 and 85 will probably also be completed this week. Nothing seems to be a blocker at the moment.

**Comment by Dominik1999** [comment]
Thanks @tohrnii so all "Must Have" Issues will be completed this week?

**Comment by Dominik1999** [comment]
@tohrnii still all good here?

**Comment by tohrnii** [comment]
@Dominik1999 Yes, the PRs for all the must haves are currently open and should be merged this week.

**Comment by grjte** [comment]
The completion of #87 has moved to #162
